addon/components/au-time-picker.js

```diff
@@ -28,103 +47,44 @@ export default class AuTimePickerComponent extends Component {
   }
 
-  /*
-   * Increments or decrements a time value.
-   * HourValue has a max of 23 while minute & seconds have a max of 59
-   * HourValue, minute & seconds have a min of 0
-   */
-
   @action
   increment(elem) {
-    if (elem == 'hourValue') {
-      ++this[elem];
-
-      if (this[elem] >= 23) {
-        this[elem] = 23;
-      }
-    }
-
-    if (elem != 'hourValue') {
-      ++this[elem];
-
-      if (this[elem] >= 59) {
-        this[elem] = 59;
-      }
-    }
+    this[elem] = this.validateTimeValue(this[elem] + 1, elem);
     this.callBackParent(this.getTimeObject);
   }
 
   @action
   decrement(elem) {
-    --this[elem];
-
-    if (this[elem] <= 0) {
-      this[elem] = 0;
-    }
+    this[elem] = this.validateTimeValue(this[elem] - 1, elem);
     this.callBackParent(this.getTimeObject);
   }
 
-  /*
-   * Triggers on keydown
-   * up arrow (38) increments the current value
-   * down arrow (40) decrements the current value
-   * If any other key that up or down has been pressed then check if it is a number key or one
-   * of the allowed keys (left/right/shift/enter..) in the keyCodes list
-   */
-
   @action
-  setTimeValue(elem, e) {
-    if (e.keyCode == 38) {
-      this.increment(elem);
-    } else if (e.keyCode == 40) {
-      this.decrement(elem);
-    } else if (
-      isNaN(parseFloat(e.key)) &&
-      this.keyCodes.indexOf(e.keyCode) == -1
-    ) {
-      e.preventDefault();
-    } else if (
-      e.target.value.length >= 2 &&
-      this.keyCodes.indexOf(e.keyCode) == -1
-    ) {
-      e.preventDefault();
+  timeValueKeyPress(type, event) {
+    switch (event.key) {
+      case 'ArrowUp':
+        this.increment(type);
+        break;
+      case 'ArrowDown':
+        this.decrement(type);
+        break;
     }
   }
 
-  /*
-   * triggered after focussing out of field. Checks if the inputted value makes sense. (e.g. hour range: 1 - 23)
-   * "elem" is the name of the tracked property
-   * "e" is the context
-   */
-
   @action
-  updateTime(elem, e) {
-    let inputValue = e.target.value;
-    if (elem == 'hourValue') {
-      if (inputValue < 0) {
-        this[elem] = 0;
-      } else if (inputValue > 23) {
-        this[elem] = 23;
-      } else {
-        this[elem] = inputValue;
-      }
-    }
-
-    if (elem != 'hourValue') {
-      if (inputValue < 0) {
-        this[elem] = 0;
-      } else if (inputValue > 59) {
-        this[elem] = 59;
-      } else {
-        this[elem] = inputValue;
-      }
-    }
+  validateTime(type, event) {
+    let tempValue = parseInt(event.target.value, 10);
+    if (isNaN(tempValue)) tempValue = 0;
+    this[type] = this.validateTimeValue(tempValue, type);
     this.callBackParent(this.getTimeObject);
   }
 
-  /*
-   * calls the function assigned to @onChange by the user with the timeObject as argument
-   */
+  validateTimeValue(value, type) {
+    const max = type === 'hourValue' ? 23 : 59;
+    value = value < 0 ? 0 : value;
+    value = value > max ? max : value;
```

Comment:
Since we use the `keyup` event this might not be super useful and the `seconds` input also makes that less ideal than I had in mind I think.

Reply:
Didn't think of that. That could centainly be done, but I also find that to be a little confusing. Imagine entering 63 seconds, would you then also want the minute value to go up by 1 and display the leftover 03 seconds? If you're not paying attention to the minute value that you have already entered, you might end up inserting the wrong time.

Reply:
This would indeed only make sense when using the arrow keys or the buttons, but it's probably not worth the complexity and requires more input from the design team. So ignore this for now, we can always tweak things in the future.